[Core] Update TPU num_visible_chips to 4 or 8 to support v5e-8

richardsliu opened the pull request:
Current `tpu.py` assumes the number of TPU visible chips can be `(1, 2, 4)`. This needs to be updated to `(1,2,4,8)`.

richardsliu committed:
Update num_visible_chips to 4 or 8

Signed-off-by: Richard Liu <39319471+richardsliu@users.noreply.github.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Update tpu.py

Signed-off-by: Richard Liu <ricliu@google.com>

ruisearch42 committed:
[ADAG] Support multi-args and kwargs in Accelerate DAG (#45545)

This PR adds multi-arg and kwarg support by serializing all positional
args and kwargs and passing it through the channel. When the channel is
read at runtime, the individual args are extracted first before passing
to the consuming tasks.

Closes #42793
---------

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>
Signed-off-by: Rui Qiao <161574667+ruisearch42@users.noreply.github.com>
Signed-off-by: Richard Liu <ricliu@google.com>

FanhaiLu1 reviewed (commented):
Thanks Richard for adding 8 chips support!

jjyao committed:
[Core] Improve doc for --object-store-memory to describe how the default value is set (#45301)

Currently it's unclear how the default value is set

Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ci] do not clean up release test data (#45627)

This code path deletes the release test working directory upon the job
completion.

We found repetitive cases where users want the data to be available for
debugging purpose. Let's rely on s3 policy to clean up the data after a
few days.

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

Commit:
Fail gracefully when cluster status is not available yet. (#45620)

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ci] remove 3.8 from ray supported python list (#45622)

Notice that we haven't removed this support completely once I work on
upgrading python 3.12.

Need to change some runtime environment to `oss-ci-base_build` since
`forge` is using python 3.8.

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

raulchen committed:
[data] Refactor resource manager (#45623)

Refactor ResourceManager and avoid it directly depending on concrete
operators.

---------

Signed-off-by: Hao Chen <chenh1024@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

ruisearch42 committed:
[Core] Fix race condition in setting node death info (#45619)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

Commit:
add env var to allow dashboard http server to serve static files with symlinks (#45618)

New env var is called RAY_DASHBOARD_BUILD_FOLLOW_SYMLINKS.

This is an advanced setting that should only be used with special Ray installations
where the dashboard build files are symlinked to a different directory.
This is not recommended for most users and can pose a security risk.
Please reference the aiohttp docs here:
https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.UrlDispatcher.add_static

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Complete do-over of RLlib release tests (new API stack). (#45589)

Signed-off-by: Richard Liu <ricliu@google.com>

hongchaodeng committed:
[core] add EC2InstanceTerminator and refactor killer creation (#45630)

Signed-off-by: hongchaodeng <hongchaodeng1@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

zcin committed:
add oss tag to container tests (#45629)

add oss tag to container tests

Add `oss` tag to container tests.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

Commit:
[HPU] [Train] Add a Stable Diffusion fine-tuning and serving example (#45217)

This PR adds an example for stable diffusion model fine-tuning and
serving using HPU. Moreover, it also covers how to adapt an existing HPU
example to run on Ray, so that users can use Ray to run the examples on
huggingface/optimum-habana.

---------

Signed-off-by: Zhi Lin <zhi.lin@intel.com>
Signed-off-by: Yunxuan Xiao <xiaoyunxuan1998@gmail.com>
Signed-off-by: Samuel Chan <116198444+anyscalesam@users.noreply.github.com>
Co-authored-by: Yunxuan Xiao <xiaoyunxuan1998@gmail.com>
Co-authored-by: Yunxuan Xiao <yunxuanx@anyscale.com>
Co-authored-by: Samuel Chan <116198444+anyscalesam@users.noreply.github.com>
Co-authored-by: Peyton Murray <peynmurray@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

Atry committed:
[Core] Handle `TypeError` when `RayTaskError.cause` is a `BaseExceptionGroup` (#45523)

Signed-off-by: Yang, Bo <bo.yang@protonbase.io>
Signed-off-by: Richard Liu <ricliu@google.com>

ruisearch42 committed:
[Core] Fix worker column off-by-one in dashboard (#45648)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

simonsays1980 committed:
[RLlib] Fix bug: Target nets are not synched with main nets in SAC. (#45614)

Signed-off-by: Richard Liu <ricliu@google.com>

Catch-Bull committed:
[Core] Fix the GIL deadlock issue caused by `list_named_actors`. (#45582)

Signed-off-by: hejialing.hjl <hejialing.hjl@bytedance.com>
Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] DreamerV3 on tf: Fix bug w/ `reduce_fn` still passed into `LearnerGroup.update_from_batch()`. (#45419)

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ci][microcheck] add a few cheap tests to microcheck (#45657)

Add keys to a few cheap builds and tests that I noticed failed on
people's PR so we can include them in microcheck. These tests are not
covered in the scope of test_in_docker.

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

jjyao committed:
[Core] Ray c++ backend structured logging (#44468)

Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

c21 committed:
[Data] Record more telemetry for newly added datasources (#45647)

This PR is to add the telemetry recording for newly added datasources.

Signed-off-by: Cheng Su <scnju13@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

ruisearch42 committed:
[Core] Expose NodeDeathInfo in state CLI (#45644)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

ruisearch42 committed:
[Core] Expose NodeDeathInfo in ActorDiedError (#45497)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

vickytsang committed:
[Core] integrate amdsmi in AMDAcceleratorManager (#44572)

Signed-off-by: root <vtsang@amd.com>
Signed-off-by: Richard Liu <ricliu@google.com>

jjyao committed:
[Core] Improve log message using structured logging (#45660)

Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

rynewang committed:
[core][bugfix] Fix the C++ GcsClient Del not respecting del_by_prefix (#45604)

Signed-off-by: Ruiyang Wang <rywang014@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

hongchaodeng committed:
[core] add chaos_many_tasks/actors terminate instance cases (#45663)

Signed-off-by: hongchaodeng <hongchaodeng1@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[core] add non-windows guard for logging tests (#45670)

`setenv` and `unsetenv` functions do not compile on windows.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

pcmoritz committed:
Make Ray compile on Python 3.12 (#45653)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Fixes https://github.com/ray-project/ray/issues/45477

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ray] python 3.12 (#45621)

Add python 3.12 wheel support. Please follow
https://docs.google.com/document/d/1-wmTLYwDA9QOHZRQ5IeUamyaTi3J0JorxvlQDbzodys/edit
for more details on the roll out phase for this support.

TL;DR; this PR will support python 3.12 as a nightly wheel; other
release artifacts will take more work

Related issue: https://github.com/ray-project/ray/issues/40211

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Co-authored-by: Philipp Moritz <pcmoritz@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

x13n committed:
[Data] Improve fractional CPU/GPU formatting (#45673)

This makes progress bar look nicer when fractional resources are used.

Signed-off-by: Daniel Kłobuszewski <daniel.klobuszewski@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Cleanup examples folder #14: Add example script for how to resume a tune.Tuner.fit() experiment from a checkpoint. (#45681)

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] DreamerV3 reporting fix for `config.report_dream_data=True`. (#45683)

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Enhance env-rendering callback. (#45682)

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ci] revert python 3.12 wheel for windows + mac (#45686)

Revert windows + mac build for python 3.12 in
https://github.com/ray-project/ray/pull/45621, they need more work

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[ci] use bazelisk to build and test on windows (#45668)

so that we can change bazel version

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

jjyao committed:
[Core] Fix test_amd_gpu.py on mac (#45688)

Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

bthananjeyan committed:
[Cluster launcher] [Azure] add option for eviction policy in azure VM template (#45397)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?
This is necessary to give users the option to pick an eviction policy
(Delete vs Deallocate) when nodes in the cluster are pre-empted.

<!-- Please give a short summary of the change and the problem this
solves. -->
This exposes an additional option that can be set in the `node_config`
attribute in the cluster YAML file.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

---
- To see the specific tasks where the Asana app for GitHub is being
used, see below:
  - https://app.asana.com/0/0/1207339070862431

Signed-off-by: bthananjeyan <brijen@ambirobotics.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Add tests

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ray] python 3.12 for macos (#45696)

Add python 3.12 wheel for macos. We need to upgrade the version of
setuptools since the previous version doesn't support python 3.12

Test:
- CI
- build: https://buildkite.com/ray-project/postmerge-macos/builds/1048

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Fix lint issue

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ray] python 3.12 for windows (#45694)

Add python 3.12 ray wheel for windows. Need to update the conda 24 which
supports python 3.12.

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

rynewang committed:
[core][runtime_env] Retry on read error in rt env agent client response read (#45513)

Signed-off-by: Ruiyang Wang <rywang014@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[core] refactor worker sys path test (#45705)

prints out `sys_path` got from remote, and reuse the package directory.

also, uses `sys.executable` instead of `python` from `PATH`, as in bazel
managed sandbox environments, these can be two different python
binaries.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

kevin85421 committed:
[Core][2/n] Core structured logging: add Text formatter and pass log config to worker process (#45344)

Signed-off-by: kaihsun <kaihsun@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

peytondmurray committed:
[Doc] Fix broken references in Ray Core documentation (#45115)

Signed-off-by: pdmurray <peynmurray@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Activate DreamerV3 weekly release test (on Pong-v5 with the 100k setup). (#45654)

Signed-off-by: Richard Liu <ricliu@google.com>

stephanie-wang committed:
[core][experimental] Add mocked tests for accelerated DAG NCCL channel (#45646)

Adds some unit tests that mock out NCCL so that we can test DAGs that use NCCL without requiring CUDA/GPUs.

Signed-off-by: Stephanie Wang <swang@cs.berkeley.edu>
Signed-off-by: Richard Liu <ricliu@google.com>

jjyao committed:
[Core] Enable logging_test on Windows (#45712)

Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

jjyao committed:
[Core] Rename ray._private.structured_logging to ray._private.ray_logging (#45717)

Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

raulchen committed:
[data] Refactor planner.py (#45706)

`planner.py` has too many if-else branches. refactor code to make the
code cleaner and easier to extend in the future.

---------

Signed-off-by: Hao Chen <chenh1024@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Add v5p and v5e

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ray] python 3.12 doc (#45703)

Add python nightly support in ray doc

Test:
-
https://anyscale-ray--45703.com.readthedocs.build/en/45703/ray-overview/installation.html

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

allenwang28 commented on python/ray/_private/accelerators/tpu.py:
```diff
@@ -33,8 +33,10 @@
 # TPU VMs come with 4 chips per host and 2 tensorcores per chip.
+# TPU v5e is a special case which has 8 chips per host.
 # For more details: https://cloud.google.com/tpu/docs/system-architecture-tpu-vm
 TPU_NUM_CHIPS_PER_HOST = 4
```
This is a bit ugly, I'm wondering if we should instead maintain a mapping between accelerator type and num chips?

Like:

```
TPU_NUM_CHIPS_PER_HOST_MAP = {
  "v5litepod-8": 8,
}
```

then below

```
num_visible_tpu_chips = len(visible_tpu_chips)
tpu_num_chips_per_host = TPU_NUM_CHIPS_PER_HOST_MAP.get(TPUAcceleratorManager.get_current_node_accelerator_type(), 4)
...
```

richardsliu replied:
I made it a static function, let me know if this works.

allenwang28 reviewed (commented):
Thanks! Just a few comments

richardsliu committed:
Fix comments

Signed-off-by: Richard Liu <ricliu@google.com>

allenwang28 commented on python/ray/_private/accelerators/tpu.py:
```diff
@@ -154,6 +154,13 @@ def validate_resource_request_quantity(
 
+    @staticmethod
+    def get_num_tpu_visible_chips_per_host(accelerator_type: str) -> int:
+      if accelerator_type == "TPU-V5LITEPOD":
```
unfortunately this isn't exactly right either, this 8 chip configuration is only specific to `v5litepod-8`.

richardsliu replied:
Fixed

richardsliu committed:
Fix pod type

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ray] build the arm64 linux wheel for python 3.12 (#45722)

Forgot to build the arm64 linux wheel for python 3.12. Fix in this PR.

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
use number of devices on node

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ci][release] add default cluster environment variable (#45707)

https://github.com/ray-project/ray/pull/40807/files mistakenly removed a
few default cluster env variable; this PR add it back

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
delete unused function

Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
delete constant

Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Update test_tpu.py

Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Rename `ParquetBaseDatasource` to `ParquetBulkDatasource` (#45728)

ParquetDatasource used to extend from ParquetBaseDatasource. However, after #40900, ParquetBaseDatasource is only used by read_bulk_datasource. To reflect this change, this PR renames ParquetBaseDatasource to ParquetBulkDatasource.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Move Parquet metadata fetching logic to separate module (#45727)

Currently, Parquet metadata fetching logic is split across parquet_datasource.py and file_meta_provider.py. Moreover, some functions defined in parquet_datasource.py are only called in file_meta_provider.py. To more coherently organize the code, this PR creates a new module parquet_meta_provider.py and moves related logic to it.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

jackhumphries committed:
Accelerated DAG: Support dynamic resizing of shared memory channels (#45323)

## Why are these changes needed?

In an accelerated DAG, all channels have their backing store specific
upfront. However, this size may not be large enough for all objects that
are communicated via channels, particularly since channels are reused
across multiple DAG invocations.

This PR adds support for increasing the size of a shared memory channel
backing store. Once the backing store size is increased, the size is
maintained (i.e., it is not decreased) until it needs to be increased
again.

Closes #45597

---------

Signed-off-by: Jack Humphries <1645405+jackhumphries@users.noreply.github.com>
Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Add "official" benchmark script for Atari PPO benchmarks (new API stack). (#45697)

Signed-off-by: Richard Liu <ricliu@google.com>

simonsays1980 committed:
[RLlib] Fix calling of callback `on_episode_created` to conform to docstring (after reset). (#45651)

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
Revert "[ci][release] add default cluster environment variable" (#45754)

Reverts ray-project/ray#45707

Signed-off-by: Richard Liu <ricliu@google.com>

allenwang28 commented on python/ray/_private/accelerators/tpu.py:
```diff
@@ -175,8 +174,11 @@ def set_current_process_visible_accelerator_ids(
 
         num_visible_tpu_chips = len(visible_tpu_chips)
-        if num_visible_tpu_chips == TPU_NUM_CHIPS_PER_HOST:
+        num_accelerators = TPUAcceleratorManager.get_current_node_num_accelerators()
```
nit: `num_accelerators_on_node` will be more descriptive

allenwang28 commented on python/ray/_private/accelerators/tpu.py:
```diff
@@ -32,9 +32,8 @@
+# By default TPU VMs come with 4 chips per host and 2 tensorcores per chip.
 # For more details: https://cloud.google.com/tpu/docs/system-architecture-tpu-vm
-TPU_NUM_CHIPS_PER_HOST = 4
 TPU_CORES_PER_CHIP = 2
```
Sorry one more thing, `TPU_CORES_PER_CHIP = 2` won't be true for v5e. We need to modify the logic [here](https://github.com/ray-project/ray/blob/0c25da892689eef8c81137044103c03ddc75e1a2/python/ray/autoscaler/_private/gcp/config.py#L127) to handle the v5litepod case.

We might be able to just remove this constant altogether and handle that in the config.py file

richardsliu replied:
Applied more fixes, please take a look.

bveeramani committed:
[Data] Merge `DefaultParquetMetadataProvider` and `ParquetMetadataProvider` (#45733)

DefaultParquetMetadataProvider is the only implementation of ParquetMetadataProvider, and we don't expect anyone (internally or externally) to add other implementations. To avoid an unnecessarily deep inheritance hierarchy, this PR removes DefaultParquetMetadataProvider and moves its implementation to ParquetMetadataProvider.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Remove unnecessary logic that manually adds partition columns (#45735)

ParquetDatasource implements logic to manually add partition values. This PR removes that logic for two reasons:
- PyArrow detects and add partition values automatically; consequently, the manual logic adds unnecessary complexity.
- The code has a bug where it doesn't respect the user-specified types.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

jjyao committed:
[Core] Introduce DictConfigProvider interface for logging config (#45736)

Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Clean up some uses of `check_num_computed` in `test_parquet.py` (#45758)

check_num_computed counts the number of read tasks executed within a LazyBlockList. With the removal of the legacy execution backend (#42780), this value is typically 0, and it's not useful to test against.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[core] print error output on `test_load_code_from_local` (#45756)

so that we know what it outputs and why it fails

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Move metadata duplication logic to Parquet metadata provider (#45734)

Currently, ParquetDatasource contains logic to deduplicate Parquet metadata. To achieve cleaner separation of responsibilities, this PR moves that logic to ParquetFileMetadataProvider.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ci][release] relax the enum of byod type (#45763)

Relax the enum of byod type; this value is basically used as the suffix
of the image, which can have a lot of different values. This will allow
more than just the ray base image to be used for running release tests.

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Apply changes for cores per chip

Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Explicitly specify metadata provider in `test_parquet_reader_estimate_data_size`  (#45766)

If you don't specify a metadata provider, Ray Data chooses a metadata provider to use; a specific metadata provider isn't guaranteed as part of the interface. The issue with test_parquet_reader_estimate_data_size is that it assumes that Ray Data chooses ParquetMetadataProvider. To avoid depending on this implementation details, this PR updates the test to explicitly pass the metadata provider.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

allenwang28 reviewed (approved):
awesome lgtm, thanks Richard! Let's make sure that we test this on a v5e when the nightly wheel is built.

bveeramani committed:
[Data] Use sampled fragments to estimate Parquet reader batch size (#45749)

ParquetDatasource uses fetched metadata to determine the Parquet reader batch size. If the metadata provider doesn't provide metadata, ParquetDatasource uses a default value of 10,000 rows. While this value might be reasonable for tabular data, the default value can lead to poor performance or even errors if each row is large. To avoid this issue, this PR updates the implementation to use the sampled fragments to determine the batch size.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Fix lint errors

Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Add `override_num_blocks` parameter to `get_parquet_metadata_provider` (#45767)

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Fix lint

Signed-off-by: Richard Liu <ricliu@google.com>

Commit:
Add train_head.py with API to serve train runs (#45711)

Signed-off-by: Alan Guo <aguo@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Fix lint

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Stats bug fix: EMA stats w/o window would lead to infinite list mem-leak. (#45752)

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Fix release test JSON summary problem (if e.g. numpy data in result dict). (#45753)

Signed-off-by: Richard Liu <ricliu@google.com>

peterghaddad committed:
[RLlib] PyFlyt QuadX WayPoints (#44758)

Signed-off-by: Richard Liu <ricliu@google.com>

saihaj committed:
docs: add ray summit banner (#45759)

Add banner for ray summit tickets.

Signed-off-by: Richard Liu <ricliu@google.com>

zcin committed:
[Core] Allow runtime env config with image_uri and container (#45765)

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Lint

Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[core] make sure PYTHONSAFEPATH is disabled (#45764)

make sure it is disabled when launching a new file/module from a local
or temp directory. newer bazel versions will enable `PYTHONSAFEPATH` by
default in python tests and binaries, and it will break ray.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

nikitavemuri committed:
[dashboard] Add search by Node ID (#45698)

Nodes don't have other identifiers that can be used to filter after they are dead, so adding search by node ID. Note: dead nodes will still get garbage collected after some time, and then won't show up in the table.

Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ci][doc/4] init api object from sphinx autodoc (#45601)

Initialize API from sphinx autodoc syntax (autoclass and autosummary).

This is part of the work to check for the consistency between API
declaration in doc vs code

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ci][doc/5] collect apis declared in sphinx autodoc (#45605)

Create an autodoc class that walks through the autodoc rst landing page,
find all child rsts and collect apis declared in those rsts.

Test:
- CI

---------

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[ci] upgrade bazel version to 6.5.0 (#45784)

and pin it to 6.5.0; rejects all other bazel version.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

Commit:
Support transfer ObjectRef from JAVA application to PYTHON deployment (#45729)

## Why are these changes needed?

In the current implementation, it's not possible to transfer an ObjectRef from a Java application to a Python deployment. For more details, please refer to this issue: https://github.com/ray-project/ray/issues/45676

**Root Cause**
In the original approach, all arguments were bundled together and passed as a list parameter, like so: {1, 2, "parameter3"}. If one of these parameters was an ObjectRef, it would lead to serialization issues, preventing cross-language transfer: {1, 2, "parameter3", objectRef4}.

**Proposed Solution**
In the new approach, each argument is passed independently. The ObjectRef is implemented as PassByReference, which avoids the previous serialization issues and enables the transfer of ObjectRef: {1, 2, "parameter3", objectRef4}.

This change enhances the interoperability between Java applications and Python deployments in Ray.

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Cleanup examples folder #10: Add `custom_rl_module.py` example script and matching RLModule example class (tiny CNN).. (#45774)

Signed-off-by: Richard Liu <ricliu@google.com>

khluu committed:
Update Docker dependencies for 2.24.0 (#45787)

Generated by release-automation bot

Signed-off-by: kevin <kevin@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[doc] change bazel version for dev workflows (#45806)

not using 5.4.1 anymore.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[release auto] allow verify wheel of a different commit (#45808)

so that we can use/test the wheel verification logic

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[cpp] use bazelisk in template example (#45805)

and use bazel 6.5.0 for recommendation

the example that we generates does not work with bazel 7.x

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

peytondmurray committed:
[Doc] Fix broken references in rllib documentation (#45168)

Signed-off-by: Richard Liu <ricliu@google.com>

shrekris-anyscale committed:
[Serve] Keep `object_store_memory` an int (#45781)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

Ray Serve converts `object_store_memory` into a float, but Ray Core
requires it to be an int. This change makes Serve keep it an int.

**Note:** I added the unit test to `test_regression.py`, but it makes
more sense to keep it in `test_api.py`. I'm going to hold off on adding
it there until https://github.com/ray-project/ray/issues/45780 is
resolved. If I add it to `test_api.py` now, the test passes and then
hangs while cleaning up.

## Related issue number

<!-- For example: "Closes #1234" -->

Closes #45321

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Shreyas Krishnaswamy <shrekris@anyscale.com>
Signed-off-by: shrekris-anyscale <92341594+shrekris-anyscale@users.noreply.github.com>
Co-authored-by: Gene Der Su <gdsu@ucdavis.edu>
Signed-off-by: Richard Liu <ricliu@google.com>

kevin85421 commented:
cc @ryanaoleary would you mind reviewing this PR?

jjyao committed:
[Core] Use exception-free json string escape for C++ structured logging (#44955)

Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

jjyao committed:
[Core] Don't add core context if Ray is not initialized (#45790)

Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[ci] fixes cuda version expansion in wanda matrix (#45809)

`froms` needs to pick the env var to track the correct base image
content for caching validation

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[ci] fix bazel version use in release automation (#45798)

don't pin bazel version globally; only pin it for cpp sanity check.
fixes nightly release automation.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
Revert "[cpp] use bazelisk in template example" (#45812)

Reverts ray-project/ray#45805

the cpp worker test does not have bazelisk available in the test
context, and it is failing

Signed-off-by: Richard Liu <ricliu@google.com>

rynewang committed:
[core] On GCS restart, load actors whose job is dead but root detached actor is alive. (#45791)

Signed-off-by: Ruiyang Wang <rywang014@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

ryanaoleary reviewed (approved):
LGTM

c21 committed:
[Data] Fix `override_num_blocks` typo of `read_lance` (#45810)

As title.

Signed-off-by: Cheng Su <scnju13@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[Tune; RLlib] Missing stopping criterion should not error (just warn). (#45613)

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Fix DreamerV3 bug for `num_env_runners > 0` (users should use `num_envs_per_env_runner > 1` instead, though). (#45819)

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Fix test case multi-agent Pendulum PPO (--num-agents=2 arg missing in BUILD). (#45820)

Signed-off-by: Richard Liu <ricliu@google.com>

angelinalg committed:
[docs] add LLM to example title for easy skimming (#45813)

Just noticed that we had trouble finding the LLM model in the Serve
examples page, but it was because we were pattern matching for `LLM`,
not `Large Language Models`.
cc: @GokuMohandas

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Signed-off-by: angelinalg <angelina@anyscale.com>
Co-authored-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

GeneDer committed:
[Serve] create internal request id to track request objects (#45761)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

User found an issue where when the request id is reused by multiple
requests around the same time, Serve's proxy will be unable to track the
request objects correctly. We should not rely on client's request id to
be unique during their lifecycle. This PR added a new
`internal_request_id` on the `RequestMetadata`, randomly generated by
the proxy, to track the request object. Client passed `request_id`
continues to be set on the `RequestMetadata` and be used for logging.
Also, updated the doc on the request id to make it more clear on what
happens when it's re-used.

## Related issue number

Closes https://github.com/ray-project/ray/issues/45723

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Gene Su <e870252314@gmail.com>
Signed-off-by: Gene Der Su <gdsu@ucdavis.edu>
Co-authored-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

BenWilson2 committed:
[Spark] Remove pricing warning when starting cluster (#45830)

Signed-off-by: Ben Wilson <benjamin.wilson@databricks.com>
Signed-off-by: Richard Liu <ricliu@google.com>

WeichenXu123 committed:
[spark] Fix error message in `_calc_mem_per_ray_node` (#45828)

Signed-off-by: Weichen Xu <weichen.xu@databricks.com>
Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[ci][release] add `RAY_BACKEND_LOG_JSON` env for all release test (#45762)

Add `RAY_BACKEND_LOG_JSON` as an cluster environment to all release
tests; requested by @jjyao

Signed-off-by: can <can@anyscale.com>
Co-authored-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

shrekris-anyscale committed:
[Serve] Remove `object_store_memory` from Ray Serve (#45814)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

This change removes the `object_store_memory` option from
`ray_actor_options`. Ray Core doesn't support setting
`object_store_memory` at the task or actor level, so it's not meaningful
to set it in a Serve deployment.

Note that #45321 has affected Serve since at least as far back as Ray
2.7, so `object_store_memory` hasn't been available in Serve recently
anyways.

## Related issue number

<!-- For example: "Closes #1234" -->

Closes #45321

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Signed-off-by: Shreyas Krishnaswamy <shrekris@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

kevin85421 commented on python/ray/autoscaler/_private/gcp/config.py:
```diff
@@ -67,7 +72,11 @@ def tpu_accelerator_config_to_type(accelerator_config: dict) -> str:
-    num_cores = num_chips * 2
+    num_cores = num_chips * DEFAULT_TPU_CORES_PER_CHIP
+
+    if generation == "v5lite_pod":
```
Why do we use `V5LITE_POD` in `accelerator_config["type"]` instead of `V5LITEPOD`? In this PR, we use both. Is it possible to make them consistent?

richardsliu replied:
Unfortunately I think this comes from an internal naming consistency in Google internal proto code. I am not sure how easy it would be to modify this.

@allenwang28

kevin85421 replied:
Thank you both for the explanation! Maybe we should at least add some comments about when to use `V5LITE_POD` and when to use `V5LITEPOD`.

richardsliu replied:
Added a comment.

aslonnie committed:
[wheel] remove py38 in build wheel script (#45840)

py38 is gone and it is not buildable

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[image] update build docker script's python version and gpu base (#45811)

using the ones that are currently being used in releases.

also removes flags that are not supported.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[wheel] provide a valid default commit (#45842)

and only build one wheel, aligning with CI's behavior.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

chris-ray-zhang committed:
Update Nav Link to be "Try Ray on Anyscale" (#45851)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Signed-off-by: Chris Zhang <chris@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[rllib] uninstall tensorflow + tensorflow_probabily before reinstall (#45838)

Some release tests are failing
https://github.com/orgs/anyscale/projects/76/views/1?pane=issue&itemId=66842882
with the error

`tensorflow.python.util.tf_export.SymbolAlreadyExposedError: Symbol
Zeros is already exposed as ().`

Some suggestions on the internet suggests the installation of tensorlfow
might be corrupted

Test:
- CI
- Release test

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Remove deprecated APIs (#45850)

This PR removes some APIs that have been "hard deprecated" (i.e., raise an error) for several releases.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Add LSTM option to `run_connector_policy`example (old API stack; w/ manual `Connector.reset()` call). (#45829)

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Cleanup examples folder #14: Add example script for policy (RLModule) inference on new API stack. (#45831)

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Fix CI test: `test_io.py`. (#45839)

Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Cleanup examples folder #15: Add example script for policy (RLModule) inference w/ ConnectorV2. (#45845)

Signed-off-by: Richard Liu <ricliu@google.com>

jackhumphries committed:
[core][experimental] Driver on different node can read from accelerated DAG output (#45786)

This PR adds support for the driver reading from the accelerated DAG
output. Previously, this was not possible unless the output writer
(i.e., the last task in the DAG) was on the same node as the driver,
which may not always be the case.

To support this, the output writer needs to be able to invoke remote
functions on the driver. This is necessary so that the output writer can
create a reader ref on the driver node, and later potentially create a
larger reader ref on the driver node if the channel backing store needs
to be resized. However, remote functions cannot be invoked on the
driver.

This PR adds a DriverActor that an accelerated DAG creates when it is
initialized. This DriverActor has an empty implementation, though it
serves as a way for the output writer to invoke remote functions on the
driver node. The workflow for creating a channel that has the driver as
a reader is this:
1. Output writer creates a writer ref on its own node.
2. Output writer invokes a remote function on DriverActor to create a
reader ref on the driver node.
3. Driver gets the channel from the output writer, which already
includes the reader ref that the DriverActor created.

The workflow for resizing a channel backing store is similar:
1. Output writer detects that the channel backing store must be resized,
so it creates a new writer ref on its own node.
2. Output writer invokes a remote function on DriverActor to create a
new reader ref on the driver node.
3. Output writer sends a `_ResizeChannel` message on the channel via the
old writer-ref/reader-ref pair. The `_ResizeChannel` message includes
the new reader ref.
4. Driver gets `_ResizeChannel` message from the output writer on the
old reader ref. The driver then updates its channel to use the new
reader ref, which is included in the `_ResizeChannel` message. Note that
the DriverActor does not read from the channel, so the driver receives
this message.

---------

Signed-off-by: Jack Humphries <1645405+jackhumphries@users.noreply.github.com>
Signed-off-by: Richard Liu <ricliu@google.com>

hongchaodeng committed:
[core] fix exit handling of FiberState threads (#45834)

Signed-off-by: hongchaodeng <hongchaodeng1@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[util] fix lint issue in inspect util (#45865)

changes the variable that is used for the iterator.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[lint] port check bazel team owner check to bazel (#45816)

so that it uses the default python version used in CI.

required to upgrade ubuntu version.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

liuxsh9 committed:
[GCS] Add `total_resources` to reduce communication overhead for aggregating alive resources (#45018)

Currently, when Ray need to collect total resources of all alive nodes,
such as selecting executable operators in Ray data, it first retrieve
the `node_table` which includes dead nodes, then filter out the alive
resources. Due to the redundant information in the node_table, this
process incurs significant overhead. When using Ray 2.9, @Bye-legumes
@nemo9cby and I observed close to 80MB/s of communication overhead while
running multiple Ray data jobs in a 100-node cluster.

[This
PR](https://github.com/ray-project/ray/pull/42817#issue-2106794703)
addresses this problem by adding `GLOBAL_LIMITS_UPDATE_INTERVAL_S` to
reduce the refresh frequency of cluster resources in Ray data. However,
reducing redundant information can further decrease communication
overhead and improve the scalability.

To achieve this, we introduce a new `TotalResources` object for
aggregating total alive resources to reduce the network overhead of
`cluster_resources` API. The reduction is shown below.

---------

Signed-off-by: Xiaoshuang Liu <liuxiaoshuang4@huawei.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[air] use isintance rather than `type()` (#45866)

fix lint error

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

peytondmurray committed:
[Doc] Fix broken references in Ray Tune documentation (#45233)

This PR fixes broken references found in the Tune documentation.

Signed-off-by: pdmurray <peynmurray@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Refactor `count()` and `schema()` to not rely on `LazyBlockList` (#45489)

Dataset.count() and Dataset.schema() implicitly depend on ExecutionPlan.execute returning a LazyBlockList.

If a Dataset is read-only, ExecutionPlan.execute doesn't perform execution and instead returns a LazyBlockList. Dataset.count() and Dataset.schema() use the metadata in the LazyBlockList to efficiently return.

To decouple count() and schema() from LazyBlockList and the implementation of execute(), this PR adds LogicalOperator.num_rows() and LogicalOperator.schema() methods. Dataset.count() and Dataset.schema() use these methods rather than LazyBlockList.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

GeneDer committed:
[Dashboard] Parse out json logs (#45853)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Most of these are taken from
[LogLineDetailDialog](https://github.com/anyscale/product/blob/ccbd8761e40790232483f5c10805020b3a733188/frontend/web/src/common/LogViewerV2/LogViewerV2.tsx#L1008)
with some modification to only show the log line itself.

This PR will allow the dashboard to only render the message part on the
log viewer if the log can be parsed as a json object. If not, then the
original log line is continue to be rendered. Also enabled the feature
to click onto a log line to render the entire log line with json log
line being parsed with pretty formatting.

## Related issue number

Closes https://github.com/ray-project/ray/issues/45869

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Also manually tested locally:
Log viewer with json logs only renders the message
<img width="1284" alt="image"
src="https://github.com/ray-project/ray/assets/7553988/feb23e1a-1860-4b72-bcd3-9fd4bbaf823c">
Clicked into a log line seeing the full parsed out json log line
<img width="1337" alt="image"
src="https://github.com/ray-project/ray/assets/7553988/fdf24a1c-46bb-411b-9615-57c510982ee2">
Log viewer with string logs render the same as before
<img width="1450" alt="image"
src="https://github.com/ray-project/ray/assets/7553988/603d5228-40fe-4035-a33c-f3f1119863f1">
Click into the log line continue to see the full message
<img width="1364" alt="image"
src="https://github.com/ray-project/ray/assets/7553988/01721a85-ad7d-4b3d-b3ce-0b9402180e6d">

---------

Signed-off-by: Gene Su <e870252314@gmail.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[ci] lint: add maxsplit key for `re.split` (#45873)

lint says it is recommended and less confusing

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[cuda] remove 11.5 and 11.6, add 12.3 (#45785)

add newer cuda versions, and stop supporting cuda versions that do not
support ubuntu 22.04

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

can-anyscale committed:
[release] remove is_byod_cluster function (#45868)

Remove `is_byod_cluster` function. Every release tests are BYOD now.

Test:
- CI

Signed-off-by: can <can@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

bveeramani committed:
[Data] Make `ExecutionPlan.execute` return `RefBundle` instead of `BlockList` (#45852)

This PR is part of a larger effort to remove `LazyBlockList`.

Currently, `ExecutionPlan.execute()` doesn't always perform execution.
Sometimes, it skips execution and returns a `LazyBlockList`. This
`LazyBlockList` has a code path separate from the standard streaming
execution code path to load data.

This PR updates the `execute()` implementation so that it always
actually perform execution, and returns a `RefBundle` rather than a
`BlockList`.

Signed-off-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: Balaji Veeramani <bveeramani@berkeley.edu>
Co-authored-by: Scott Lee <scottjlee@users.noreply.github.com>
Signed-off-by: Richard Liu <ricliu@google.com>

sven1977 committed:
[RLlib] Cleanup examples folder #13. Fix main examples docs page for RLlib. (#45382)

Signed-off-by: Richard Liu <ricliu@google.com>

aslonnie committed:
[ci] avoid deadlocking bazel query and bazel run (#45892)

in team attribute checking

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
add comment

Signed-off-by: Richard Liu <ricliu@google.com>

richardsliu committed:
Merge branch 'master' into patch-1

richardsliu committed:
Merge branch 'master' into patch-1

richardsliu committed:
Merge branch 'master' into patch-1

jjyao commented on python/ray/autoscaler/_private/gcp/config.py:
```diff
@@ -106,7 +118,7 @@ def _validate_tpu_config(node: dict):
         topology_pattern = re.compile(r"^\d+x\d+(x\d+)?$")
 
-        if not generation_pattern.match(generation):
+        if generation != "V5LITE_POD" and not generation_pattern.match(generation):
```
Is this case sensitive?

richardsliu replied:
The original string from the proto should always be all caps.
